controller/tests/test_plan_controller.py

```diff
@@ -0,0 +1,147 @@
+                    camps=[Camp(name='TestCamp')])
+        close_plan(plan)
+        self.assertEqual(date.today(), plan.close_date)
+
+    def test_find_plan_controller(self):
+        """
+        Test to see find plan given plan name.
+        """
+        test_plan = Plan(name='My Plan',
+                         emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                         description='Test emergency plan',
+                         geographical_area='',
+                         camps=[Camp(name='TestCamp')])
+        found_plan = find_plan('My Plan')
+        self.assertEqual(test_plan, found_plan)
+        self.setUp()
+
+    def test_no_plan_found_controller(self):
+        """
+        Test that if no plan is found then an exception is raised.
+        """
+        with self.assertRaises(controller_error.ControllerError):
+            find_plan('My Plan')
+
+        self.setUp()
+
+    def test_not_plan_instance(self):
+        """
+        Test that if input is not a Plan type then exception is raised.
+        """
+        camp_1 = Camp(name='TestCamp')
+        with self.assertRaises(controller_error.ControllerError):
+            find_plan(camp_1.name)
+
+        self.setUp()
+
+    def test_create_plan_controller(self):
+        """
+        Test to see if plan is created correctly with controller.
+        """
+        plan = create_plan(plan_name='My Plan',
+                           emergency_type=Plan.EmergencyType.EARTHQUAKE,
+                           description='Test emergency plan',
+                           geographical_area='',
+                           camps=[Camp(name='TestCamp')])
+        self.assertEqual('My Plan', plan.name)
+        plan_camp = plan.camps.get('TestCamp')
+        self.assertListEqual([plan_camp.name], ['TestCamp'])
+        self.setUp()
+
+    def test_create_camps_controller(self):
+        """
+        Test that camps are added to plan correctly.
+        """
+        create_camps('Camp 1')
+        self.assertIsInstance(create_camps('Camp 1'), Camp)
+        self.setUp()
+
+    def test_list_plans_controller(self):
+        """
+        Test to see if plans are listed correctly.
+        """
+        Plan(name='First Plan',
+             emergency_type=Plan.EmergencyType.EARTHQUAKE,
+             description='Test emergency plan',
+             geographical_area='',
+             camps=[Camp(name='TestCamp')])
+
+        Plan(name='Second Plan',
+             emergency_type=Plan.EmergencyType.EARTHQUAKE,
+             description='Test emergency plan',
+             geographical_area='',
+             camps=[Camp(name='TestCamp')])
+
+        Plan(name='Third Plan',
+             emergency_type=Plan.EmergencyType.EARTHQUAKE,
+             description='Test emergency plan',
+             geographical_area='',
+             camps=[Camp(name='TestCamp')])
+
+        find_list = [Plan.find('First Plan'), Plan.find('Second Plan'), Plan.find('Third Plan')]
```

Comment:
We typically name the comparison values for tests `expected`
so `expected = [...]`

Reply:
I see, I'll change it now.